src/action.ml

```diff
@@ -69,38 +69,45 @@ struct
   let rec sexp_of_t : _ -> Sexp.t =
     let path = Path.sexp_of_t and string = String.sexp_of_t in
     function
-    | Run (a, xs) -> List (Atom "run" :: Program.sexp_of_t a :: List.map xs ~f:string)
-    | Chdir (a, r) -> List [Atom "chdir" ; path a ; sexp_of_t r]
-    | Setenv (k, v, r) -> List [Atom "setenv" ; string k ; string v ; sexp_of_t r]
+    | Run (a, xs) -> List (Sexp.unsafe_atom_of_string "run"
+                           :: Program.sexp_of_t a :: List.map xs ~f:string)
+    | Chdir (a, r) -> List [Sexp.unsafe_atom_of_string "chdir" ;
+                            path a ; sexp_of_t r]
+    | Setenv (k, v, r) -> List [Sexp.unsafe_atom_of_string "setenv" ;
+                                string k ; string v ; sexp_of_t r]
     | Redirect (outputs, fn, r) ->
-      List [ Atom (sprintf "with-%s-to" (Outputs.to_string outputs))
+      List [ Quoted_string (sprintf "with-%s-to" (Outputs.to_string outputs))
            ; path fn
            ; sexp_of_t r
            ]
     | Ignore (outputs, r) ->
-      List [ Atom (sprintf "ignore-%s" (Outputs.to_string outputs))
+      List [ Quoted_string (sprintf "ignore-%s" (Outputs.to_string outputs))
```

Comment:
This should stay an atom

Reply:
So `outputs` is guaranteed to never break the Atom invariant?

Reply:
`Outputs.to_string outputs` is one of `"stdout"`, `"stderr"` or `"outputs"`